pallets/anchor/src/tests.rs

```diff
@@ -139,3 +149,120 @@ fn should_be_able_to_add_neighbors_and_check_history() {
 		);
 	});
 }
+
+#[test]
+fn anchor_works() {
+	new_test_ext().execute_with(|| {
+		const M: usize = 2;
+		type Bn254Fr = ark_bn254::Fr;
+		let mut rng = ark_std::test_rng();
+		let curve = Curve::Bn254;
+		let params3 = setup_circom_params_x5_3::<Bn254Fr>(curve);
+		let params5 = setup_circom_params_x5_5::<Bn254Fr>(curve);
+		// init hasher pallet first.
+		assert_ok!(HasherPallet::force_set_parameters(Origin::root(), params3.to_bytes()));
+		// then the merkle tree.
+		<MerkleTree as OnInitialize<u64>>::on_initialize(1);
+		// now let's create the anchor.
+		let deposit_size = 10_000;
+		let max_edges = 2;
+		let depth = LEN as u8;
+		let asset_id = 0;
+		assert_ok!(Anchor::create(Origin::root(), deposit_size, max_edges, depth, asset_id));
+
+		// now with anchor created, we should setup the circuit.
+		let tree_id = MerkleTree::next_tree_id() - 1;
+		let src_chain_id = 1;
+		let sender_account_id = 1;
+		let recipient_account_id = 2;
+		let relayer_account_id = 0;
+		let fee_value = 0;
+		let refund_value = 0;
+		// top-up the sender account with some balance.
+		let _ = Balances::set_balance(Origin::root(), sender_account_id, 100_000_000, 0);
+		// inputs
+		let chain_id = Bn254Fr::from(src_chain_id);
+		let recipient = Bn254Fr::from(recipient_account_id);
+		let relayer = Bn254Fr::from(relayer_account_id);
+		let fee = Bn254Fr::from(fee_value);
+		let refund = Bn254Fr::from(refund_value);
+		let (leaf_private, leaf_public, leaf, nullifier_hash) = setup_leaf_circomx5(chain_id, &params5, &mut rng);
+		let leaf_element = Element::from_bytes(&leaf.into_repr().to_bytes_le());
+		let nullifier_hash_element = Element::from_bytes(&nullifier_hash.into_repr().to_bytes_le());
+		assert_ok!(Anchor::deposit(
+			Origin::signed(sender_account_id),
+			tree_id,
+			leaf_element
+		));
+		let tree_root = MerkleTree::get_root(tree_id).unwrap();
+		let height = 1;
+		// add edges
+		assert_ok!(<Anchor as AnchorInterface<_>>::add_edge(
```

Comment:
Why do you add an edge for this test? It doesn't seem like it uses 2 anchors?

Reply:
I will **move** it to another test where we can test cross Anchor withdrawal later on.